pkg/storage/chunk/aws/dynamodb_storage_client.go

```diff
@@ -113,6 +113,10 @@ type dynamoDBStorageClient struct {
 	metrics *dynamoDBMetrics
 }
 
+func (a dynamoDBStorageClient) AsyncQueueLength() int {
```

Comment:
I would suggest not adding this method which is only relevant for caching index client and has no use beyond just test.

Reply:
fix

type IndexAsyncClient interface {
	AsyncQueueLength() int
}

			assert.Eventually(t, func() bool {
				if asyncClient, ok := client.(IndexAsyncClient); ok {
					return asyncClient.AsyncQueueLength() == 0
				}
				return true
			}, time.Second, 10*time.Millisecond)